tests/src/buildApplication/generateFunctionsMap.test.ts

```diff
@@ -1,105 +1,75 @@
-			},
-			readdir: async (rawDir: string) => {
-				const dir = normalizePath(rawDir);
-				if (['validTest/functions', 'invalidTest/functions'].includes(dir)) {
-					return [
-						'api',
-						'index.func',
-						'index.rsc.func',
-						'middlewarejs.func',
-						'base/middleware.func',
-						'path/(group-1)/to/(group-2)/page.func',
-						'(is-valid)/should-be-valid.func', // valid
-						'should-be-valid.func', // invalid
-						'should-be-valid-alt.func', // invalid
-						'rsc/(is-valid)/should-be-valid.func', // valid
-						'rsc/should-be-valid.rsc.func', // invalid
-					];
-				}
-				if (
-					['validTest/functions/api', 'invalidTest/functions/api'].includes(dir)
-				) {
-					return ['hello.func'];
-				}
-				return [];
-			},
-		};
-	});
+const validIndexVcConfigJson = JSON.stringify({
+	runtime: 'edge',
+	entrypoint: 'index.js',
 });
-
-afterAll(() => {
-	vi.clearAllMocks();
+const invalidIndexVcConfigJson = JSON.stringify({
+	runtime: 'nodejs',
+	entrypoint: 'index.js',
 });
 
+const validFuncDir = {
+	'.vc-config.json': validIndexVcConfigJson,
+	'index.js': '',
+};
+
+const invalidFuncDir = {
+	'.vc-config.json': invalidIndexVcConfigJson,
+	'index.js': '',
+};
+
 describe('generateFunctionsMap', async () => {
 	test('should generate a valid functions map (without experimentalMinify), accounting for invalid root-level functions', async () => {
-		const { invalidFunctions, functionsMap } = await generateFunctionsMap(
-			'validTest/functions',
-			false
-		);
+		mockFs({
+			functions: {
+				'index.func': validFuncDir,
+				'index.rsc.func': validFuncDir,
+				api: {
+					'hello.func': validFuncDir,
+				},
+				'middlewarejs.func': {
+					'.vc-config.json': JSON.stringify({
+						name: 'middleware',
+						runtime: 'edge',
+						entrypoint: 'middleware.js',
+					}),
+					'index.js': '', // is this correct? @james-elicx
+				},
+				base: {
+					'middleware.func': validFuncDir,
+				},
+				path: {
+					'(group-1)': {
+						to: {
+							'(group-2)': {
+								'page.func': validFuncDir,
+							},
+						},
+					},
+				},
+				'(is-valid)': {
+					'should-be-valid.func': validFuncDir,
+				},
+				rsc: {
+					'(is-valid)': {
+						'should-be-valid.func': validFuncDir,
+						'should-be-valid.rsc.func': validFuncDir,
+					},
+				},
+			},
+		});
 
-		// NOTE: The invalid function here is used to test that invalid functions on the root-level are considered invalid, while a valid squashed function (in a route group) replaces an equivalent invalid function that exists on the root-level.
-		// i.e. `(is-valid)/should-be-valid.func` replaces the invalid `should-be-valid.func` on the root-level, and `should-be-valid-alt.func` is still invalid.
-		expect(invalidFunctions.size).toEqual(1);
-		expect(Array.from(invalidFunctions.values())).toEqual([
-			'should-be-valid-alt.func',
-		]);
+		const { functionsMap } = await generateFunctionsMap('functions', false);
+
+		// Do we still need to test this? @james-elicx
```

Comment:
So the reason I had this was to verify that route squashing was working properly. There was a root file that was invalid that wouldn't be overridden by squashing routes, and another that was invalid that would be overridden by squashing routes. This test would verify that the one without a path inside a group would be invalid, while the one expected to be overridden is valid.

At the moment, I don't believe such a case is being tested anymore, it is instead just being assumed. Probably worth adding a separate test just to make sure this expected behaviour is actually happening.

Input:
- should-be-valid.func -> invalid
- should-be-valid-alt.func -> invalid
- (is-valid)/should-be-valid.func -> valid

Expected output:
- should-be-valid.func -> valid as it was overridden by squashing a route group.
- should-be-valid-alt.func -> invalid as there was no other route to override it.